src/Controls/src/Core/Platform/AlertManager/DialogHelper.Gtk.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Linq;
+using GLib;
+using Gtk;
+using Microsoft.Maui.Controls.Internals;
+
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	// Ported from https://github.com/nblockchain/DotNetForms/blob/5.0.0/Xamarin.Forms.Platform.GTK/Helpers/DialogHelper.cs
+	internal static class DialogHelper
+	{
+		public static void ShowAlert(Gtk.Window window, AlertArguments arguments)
+		{
+			MessageDialog messageDialog = new MessageDialog(
+					window,
+					DialogFlags.DestroyWithParent | DialogFlags.UseHeaderBar,
+					MessageType.Other,
+					GetAlertButtons(arguments),
+					arguments.Message);
+
+			SetDialogTitle(arguments.Title, messageDialog);
+			SetButtonText(arguments.Accept, ResponseType.Ok, messageDialog);
+			SetButtonText(arguments.Cancel, ResponseType.Cancel, messageDialog);
+
+			ResponseType result = (ResponseType)messageDialog.Run();
+
+			if (result == ResponseType.Ok)
+			{
+				arguments.SetResult(true);
+			}
+			else
+			{
+				arguments.SetResult(false);
+			}
+
+			messageDialog.Destroy();
+		}
+
+		public static void ShowActionSheet(Gtk.Window window, ActionSheetArguments arguments)
+		{
+			// as title bar is not shown, use dialog body for title text
+
+			MessageDialog messageDialog = new MessageDialog(
+			   window,
+			   DialogFlags.DestroyWithParent | DialogFlags.UseHeaderBar,
+			   MessageType.Other,
+			   ButtonsType.Cancel,
+			   arguments.Title);
+
+
+			//SetDialogTitle(arguments.Title, messageDialog);
+			SetButtonText(arguments.Cancel, ResponseType.Cancel, messageDialog);
+			SetDestructionButton(arguments.Destruction, messageDialog);
+			AddExtraButtons(arguments, messageDialog);
+
+			int result = messageDialog.Run();
+
+			if ((ResponseType)result == ResponseType.Cancel)
```

Comment:
@webwarrior-ws let's do the cast in the result assignment (micro optimization I know...)